docs/config.md

```diff
@@ -1,100 +1,272 @@
-# Config system
+- [Config](#config)
+  - [Basic](#basic)
+  - [Interpolation](#interpolation)
+    - [Callable Interpolation](#callable-interpolation)
+    - [Interpolation by Path](#interpolation-by-path)
+    - [Interpolation by Attribute](#interpolation-by-attribute)
+    - [Default Value by Path Interpolation](#default-value-by-path-interpolation)
+    - [Ignore Interpolation](#ignore-interpolation)
+    - [Lazy Evaluation](#lazy-evaluation)
 
-Algorithm configurations such as training hyperparameters are serialized as nested mappings often using the YAML format.  
-Nested mappings are parsed by `pytorch_pfn_extras.config.Config` that returns a dictionary of objects that are lazily evaluated.
-The most notable feature is a special handling of the `type` key, which invokes a function or instantiates an object. The arguments for them are mapped by non-`type` keys.
-```python 
-import yaml
+
+# Config
+
+## Basic
+Config object is created from a dictionary.
+
+```python
 from pytorch_pfn_extras.config import Config
-from torchvision.models.resnet import resnet50
+import yaml
+pre_eval_config = yaml.load('''
+foo:
+  bar: 'bar_value'
+  ls:
+    - 'first'
+    - key0: 'value0'
+      key1: 'value1'
+baz: 'baz_value'
+''')
+config = Config(pre_eval_config)
+```
 
-types = {
-  'resnet50': resnet50
-}
+Accessing config values:
+```python
+print(config['/foo/ls/0'])
+# 'first'
+print(config['/foo/ls/1/key0'])
+# 'value0'
+print(config['/foo/ls'])
+# ['first', {'key0': 'value0', 'key1': 'value1'}]
+print(config['/baz'])
+# 'baz_value'
+```
+
+## Interpolation
+
+### Callable Interpolation
+
+You could replace a value as the return value of a callable.
+- `types` is an additional input to `Config`. `types` is a mapping from a callable's name to the actual callable.
+- A sub-dictionary containing the key `type` invokes callable interpolation.
+
+```python
 pre_eval_config = yaml.load('''
-model:
-  type: resnet50
-  num_classes: 10
+name:
+  type: concat
+  x0: 'First'
+  x1: 'Last'
 ''')
+
+types = {
+  'concat': lambda x0, x1: x0 + ' ' + x1
+}
+
 config = Config(pre_eval_config, types)
-resnet = config['/model']  # same as resnet50(num_classes=10)
+# the value returned by
+# concat(x0='First', x1='Last')
+print(config['/name'])
+# 'First Last'
+```
+
+#### How it works
```

Comment:
Do we need this section? `resolve_key` is inaccurate and not helpful to understand the behavior. I think `the value returned by concat(x0='First', x1='Last')` (in the example) is enough to explain.